runtime.go

```diff
@@ -63,6 +64,14 @@ type _runtime struct {
 
 func (self *_runtime) enterScope(scope *_scope) {
 	scope.outer = self.scope
+	if self.scope != nil {
+		scope.depth = self.scope.depth + 1
+	}
+
+	if self.stackLimit != 0 && scope.depth >= self.stackLimit {
+		panic(self.panicRangeError("RangeError: Maximum call stack size exceeded"))
```

Comment:
I did think it was a bit fishy. I'm happy to change this, and perhaps the others as well, so long as they're still identifiable as RangeError to the casual observer of a JS stack trace.